File: src/Entity/Produit.php
<?php

declare(strict_types=1);

namespace App\Entity;

use App\Repository\ProduitRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ProduitRepository::class)]
#[ORM\Table(name: 'produit')]
class Produit
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private string $title = '';

    #[ORM\Column(length: 50, options: ['default' => 'medical'])]
    private string $category = 'medical';

    #[ORM\Column]
    private float $price = 0.0;

    #[ORM\Column]
    private float $tva = 0.0;

    #[ORM\Column(length: 255, nullable: true)]
    private ?string $image = null;

    #[ORM\Column(type: 'text', nullable: true)]
    private ?string $description = null;

    #[ORM\Column]
    private int $stock = 0;

    #[ORM\ManyToOne(inversedBy: 'produits')]
    #[ORM\JoinColumn(nullable: true, onDelete: 'SET NULL')]
    private ?User $owner = null;

    /**
     * @var Collection<int, Panier>
     */
    #[ORM\OneToMany(mappedBy: 'produit', targetEntity: Panier::class)]
    private Collection $paniers;

    public function __construct()
    {
        $this->paniers = new ArrayCollection();
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getTitle(): string
    {
        return $this->title;
    }

    public function setTitle(string $title): static
    {
        $this->title = trim($title);

        return $this;
    }

    public function getCategory(): string
    {
        return $this->category;
    }

    public function setCategory(string $category): static
    {
        $normalized = strtolower(trim($category));
        if (!array_key_exists($normalized, self::allowedCategories())) {
            $normalized = 'medical';
        }

        $this->category = $normalized;

        return $this;
    }

    /**
     * @return array<string, string>
     */
    public static function allowedCategories(): array
    {
        return [
            'medical' => 'Medical',
            'clothing' => 'Clothing',
            'toys' => 'Toys',
            'food' => 'Food',
        ];
    }

    public function getCategoryLabel(): string
    {
        $categories = self::allowedCategories();

        return $categories[$this->getCategory()] ?? ucfirst($this->getCategory());
    }

    public function getPrice(): float
    {
        return $this->price;
    }

    public function setPrice(float $price): static
    {
        $this->price = $price;

        return $this;
    }

    public function getTva(): float
    {
        return $this->tva;
    }

    public function setTva(float $tva): static
    {
        $this->tva = $tva;

        return $this;
    }

    public function getImage(): ?string
    {
        return $this->image;
    }

    public function setImage(?string $image): static
    {
        $this->image = $image !== null ? trim($image) : null;

        return $this;
    }

    public function getDescription(): ?string
    {
        return $this->description;
    }

    public function setDescription(?string $description): static
    {
        $this->description = $description !== null ? trim($description) : null;

        return $this;
    }

    public function getStock(): int
    {
        return $this->stock;
    }

    public function setStock(int $stock): static
    {
        $this->stock = $stock;

        return $this;
    }

    public function getOwner(): ?User
    {
        return $this->owner;
    }

    public function setOwner(?User $owner): static
    {
        $this->owner = $owner;

        return $this;
    }

    /**
     * @return Collection<int, Panier>
     */
    public function getPaniers(): Collection
    {
        return $this->paniers;
    }

    public function addPanier(Panier $panier): static
    {
        if (!$this->paniers->contains($panier)) {
            $this->paniers->add($panier);
            $panier->setProduit($this);
        }

        return $this;
    }

    public function removePanier(Panier $panier): static
    {
        $this->paniers->removeElement($panier);

        return $this;
    }
}

File: modules/products/java/com/projet/entities/Produit.java
package com.projet.entities;

public class Produit {
    // attributs
    private int id;
    private String title;
    private double price;
    private double tva;
    private String image;
    private String description;
    private int stock;

    // constructeur vide
    public Produit() {}

    // constructeur sans id
    public Produit(String title, double price, double tva, String image, String description, int stock) {
        this.title = title;
        this.price = price;
        this.tva = tva;
        this.image = image;
        this.description = description;
        this.stock = stock;
    }

    // constructeur avec id
    public Produit(int id, String title, double price, double tva, String image, String description, int stock) {
        this.id = id;
        this.title = title;
        this.price = price;
        this.tva = tva;
        this.image = image;
        this.description = description;
        this.stock = stock;
    }

    // getters & setters
    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public double getTva() {
        return tva;
    }

    public void setTva(double tva) {
        this.tva = tva;
    }

    public String getImage() {
        return image;
    }

    public void setImage(String image) {
        this.image = image;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public int getStock() {
        return stock;
    }

    public void setStock(int stock) {
        this.stock = stock;
    }

    @Override
    public String toString() {
        return "Produit{" +
                "id=" + id +
                ", title='" + title + '\'' +
                ", price=" + price +
                ", tva=" + tva +
                ", image='" + image + '\'' +
                ", description='" + description + '\'' +
                ", stock=" + stock +
                '}';
    }
}

